--- origsrc/abraca-0.4.3/site_scons/site_tools/binaryblob.py	2010-01-06 15:57:09.000000000 -0600
+++ src/abraca-0.4.3/site_scons/site_tools/binaryblob.py	2010-02-08 23:54:12.180992500 -0600
@@ -18,13 +18,11 @@ import os
 
 
 _assembly_template = """
-  .section ".rodata"
-  .globl %(full_name)s
-  .type %(full_name)s, %%object
-%(full_name)s:
+  .section .rodata
+  .globl _%(full_name)s
+_%(full_name)s:
   .incbin "%(file)s" /* MD5: %(md5)s */
   .byte 0
-  .size %(full_name)s, .-%(full_name)s
 """
 
 
