From 665a12fa5f34423c32030ad94a9c2e1c0f782e77 Mon Sep 17 00:00:00 2001
From: Daniel Svensson <dsvensson@gmail.com>
Date: Sun, 23 Jan 2011 21:36:04 +0000
Subject: OTHER: Compile fixes for valac 0.12 track.

---
diff --git a/SConstruct b/SConstruct
index 831e343..637f1ab 100644
--- a/SConstruct
+++ b/SConstruct
@@ -5,7 +5,7 @@ env.VariantDir('build', '.')
 env.Append(VALAPKGPATH = ['vapi'])
 
 conf = env.Configure('build/build-config.h')
-conf.CheckVala('0.10.0')
+conf.CheckVala('0.11.5')
 conf.CheckCCompiler()
 conf.CheckPkgConfig()
 conf.CheckApp('msgfmt')
diff --git a/src/components/filter/filter_view.vala b/src/components/filter/filter_view.vala
index e0efdba..f3af402 100644
--- a/src/components/filter/filter_view.vala
+++ b/src/components/filter/filter_view.vala
@@ -370,7 +370,9 @@ namespace Abraca {
 							} else {
 								order = Gtk.SortType.DESCENDING;
 							}
-							sorting = {column.title, order};
+							/* TODO: This temp variable is needed due to a bug in vala 0.11.5 */
+							Sorting tmp = {column.title, order};
+							sorting = tmp;
 							break;
 						}
 					}
diff --git a/src/coverart.vala b/src/coverart.vala
index 9e4959e..a6b55f3 100644
--- a/src/coverart.vala
+++ b/src/coverart.vala
@@ -412,7 +412,7 @@ namespace Abraca {
 			if (!get_track_ids (candidates, pixbuf, out ids))
 				return true;
 
-			char[] buffer;
+			uint8[] buffer;
 			if (!pixbuf.save_to_buffer (out buffer, "png"))
 				return true;
 
@@ -420,7 +420,7 @@ namespace Abraca {
 			/* TODO: Need to store ids here due to a Vala reference bug. */
 			selected_ids = ids;
 
-			client.xmms.bindata_add ((uint8[]) buffer).notifier_set (on_bindata_add);
+			client.xmms.bindata_add (buffer).notifier_set (on_bindata_add);
 
 
 			return true;
diff --git a/src/toolbar.vala b/src/toolbar.vala
index 5b20dad..b2d6a52 100644
--- a/src/toolbar.vala
+++ b/src/toolbar.vala
@@ -343,7 +343,7 @@ namespace Abraca {
 
 			try {
 				var loader = new Gdk.PixbufLoader();
-				loader.write(data, data.length);
+				loader.write(data);
 				loader.close();
 
 				var pixbuf = loader.get_pixbuf();
--
cgit v0.8.3.4
